Foundation/FileManager.swift

```diff
@@ -2377,46 +2377,113 @@ extension FileManager {
-        internal typealias ErrorHandler = /* @escaping */ (URL, Error) -> Bool
+#if os(Windows)
+        var _url : URL
+        var _options : FileManager.DirectoryEnumerationOptions
+        var _errorHandler : ((URL, Error) -> Bool)?
+        var _stack: [URL]
+        var _current: URL?
+        var _root_depth : Int
 
-        init(url: URL, options: FileManager.DirectoryEnumerationOptions, errorHandler: ErrorHandler?) {
+
+        init(url: URL, options: FileManager.DirectoryEnumerationOptions, errorHandler: (/* @escaping */ (URL, Error) -> Bool)?) {
+            _url = url
+            _options = options
+            _errorHandler = errorHandler
+            _stack = [url]
+            _root_depth = url.pathComponents.count
+        }
+
+
+        override func nextObject() -> Any? {
+            func contentsOfDir(directory: URL) -> [URL]? {
+                var ffd: WIN32_FIND_DATAW = WIN32_FIND_DATAW()
+                let dirPath = Array((directory.absoluteString + "\\*\0").utf16)
+                guard let result = dirPath.withContiguousStorageIfAvailable(
+                    {FindFirstFileW($0.baseAddress, &ffd)}) else {
+                    return nil
+                }
+                guard let h = result, h != INVALID_HANDLE_VALUE else {
+                    return nil
+                }
+
+                var files: [URL] = []
+                repeat {
+                    let fileArr = [WCHAR](UnsafeBufferPointer(start: &ffd.cFileName.0,
+                                                              count: MemoryLayout.size(ofValue: ffd.cFileName)))
+                    let file = String(decodingCString: fileArr, as: UTF16.self)
+                    if (file != "."
+                        && file != ".."
+                        && (_options.contains(.skipsHiddenFiles)
+                            && (ffd.dwFileAttributes & DWORD(FILE_ATTRIBUTE_HIDDEN) == 0))) {
+                        files.append(URL(fileReferenceLiteralResourceName: "\(directory.absoluteString)\\\(file)"))
+                      }
+                } while(FindNextFileW(h, &ffd) != 0)
+                return files
+            }
+            while let url = _stack.popLast() {
+                if url.hasDirectoryPath && !_options.contains(.skipsSubdirectoryDescendants) {
+                    guard let dirContents = contentsOfDir(directory: url)?.reversed() else {
+                        let keepGoing: Bool
+                        if let handler = _errorHandler {
+                            keepGoing = handler(URL(fileURLWithPath: url.absoluteString),
+                                                _NSErrorWithWindowsError(GetLastError(), reading: true))
+                        } else {
+                            keepGoing = true
+                        }
+                        if !keepGoing {
+                            return nil
+                        }
+                        continue
```

Comment:
Could this be collapsed down to 
```
if let handler = _errorHandler {
    let keepGoing = handler(URL(fileURLWithPath: url.absoluteString),
	          _NSErrorWithWindowsError(GetLastError(), reading: true))
    if !keepGoing { return nil }
}
continue
```

Reply:
I think that the unwrapped version that @spevans suggested is easier to read.